### ISSUE TITLE
Reproduce and fix failures on the ATLAS/Fedora testing platform

### ISSUE THREAD

**Issue by jwbowers** [markmfredrickson/RItools#128]
We have learned that our approach to warning users about singularities is producing errors on the ATLAS/Fedora testing plaform. See here: https://www.stats.ox.ac.uk/pub/bdr/Rblas/ATLAS/RItools.out 

RItools is currently removed from CRAN.

Mark is working on creating a version of that testing suite in docker so that we can learn more about the source of the error.

**Comment by markmfredrickson** [comment]
So far I have been unable to recreate this warning.

My first attempt used the docker images from [rocker](https://rocker-project.org) because they have pre-compiled binaries for linux. If I could recreate the issue, having pre-compiled binaries would save a lot of time. These are, however, based on ubuntu, so the setting wouldn't be exactly the same. In any case, I could not preproduce the error when using the ATLAS libraries.

My [second attempt was based on Fedora 34](https://gist.github.com/markmfredrickson/46f7c2e3588b823a1f0831451653b2ba) using the precompiled version of R for that platform (4.0.2). This binary is linked to the flexiblas library, which makes it easy to select other backends. I can verify that ATLAS is the backend being used from with R using the `rflexiblas` library, but again, no warning.

They provide two descriptions of how the environment is set up:

1. [The compilation of R itself](https://www.stats.ox.ac.uk/pub/bdr/Rconfig/r-devel-linux-x86_64-fedora-gcc), along with several other libraries not drawn from the Fedora 34 package system.
2. [Configuration of BLAS libraries](https://www.stats.ox.ac.uk/pub/bdr/Rblas/README.txt)

I was hoping to avoid all the configuration and compilation in first step, but at this point I'm not sure if there is much else to do than just make that happen.

**Comment by jwbowers** [comment]
FYI here are notes on the platforms where the errors were emerging: https://www.stats.ox.ac.uk/pub/bdr/Rblas/README.txt

**Comment by jwbowers** [comment]
@markmfredrickson Is there anything else we can do? Do you think that the symlinking described in <https://www.stats.ox.ac.uk/pub/bdr/Rblas/README.txt> produce results that differ from using `rflexiblas`?

**Comment by benthestatistician** [comment]
I think that Nick HD compiles R from scratch.  Configuration of BLAS libraries also sounds to be up his alley. We could ask him whether he'd be willing to take these things on, either in a docker image or perhaps just on another some machine he controls alongside of the regular R configuration. 

The IES-TEA project has time for him budgeted, particularly if this doesn't add up to an unreasonable number of hours. I bet he'd enjoy hearing from you also, @jwbowers.  Whether he'd have space for this in his schedule is an entirely different question, of course.

**Comment by markmfredrickson** [comment]
> @markmfredrickson Is there anything else we can do? Do you think that the symlinking described in https://www.stats.ox.ac.uk/pub/bdr/Rblas/README.txt produce results that differ from using `rflexiblas`?
> 

There does not appear to be an `libRblas.so` when using the pre-compiled R binaries on Fedora 34, so the symlink idea would not appear to apply.

> Or using R-devel?

The `R-devel` package on Fedora is just the same as base R for that platform. We could go with the source compile approach, but I was trying to avoid that if possible due to the lengthy dependency issues listed in the setup description from CRAN.

> I think that Nick HD compiles R from scratch. Configuration of BLAS libraries also sounds to be up his alley. We could ask him whether he'd be willing to take these things on, either in a docker image or perhaps just on another some machine he controls alongside of the regular R configuration.

First, I'm sure he's better at these things than I am. And second, if he has time this week, that would solve the additional problem that I'm tied up Wed - Fri of this week. My vote would be for him to use whatever system would be fastest for him, provided we can do some debugging on the package after.

**Comment by benthestatistician** [comment]
(I'd be quite surprised if Nick were in a position to take this on this week.  I'd still support asking if he could fit it into his queue.)

**Comment by jwbowers** [comment]
My concern is that Nick might not be able to exactly replicate the system
that was described since Mark got quite close but didn't have luck.

Jake Bowers
http://jakebowers.org

**Comment by markmfredrickson** [comment]
I found another [docker image from rhub](https://hub.docker.com/r/rhub/fedora-gcc-devel) that could be useful. It is based on Fedora 33 (not 34), but it does come with yesterday's R-devel. I installed the F33 version of ATLAS, which has the same version string (3.10.3). I then symlinked the `libRblas.so` to the ATLAS version. Here's R's `sessionInfo()` showing these facts:

```
> sessionInfo()
R Under development (unstable) (2022-08-08 r82696)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Fedora 33 (Container Image)

Matrix products: default
BLAS/LAPACK: /usr/lib64/atlas/libsatlas.so.3.10
```

I am still unable, however, to recreate the warning. Either running the test suit and running the code directly. There is something very particular about this setup that is generating the warning, and I'm not sure what it is.

More recent [Dockerfile](https://gist.github.com/markmfredrickson/7be427c83402ee78fb4f3cae90eef869)

To use, download,

```
$ cd LOCATION_OF_DOCKERFILE
$ docker build -t ritools-atlas .
$ docker run --rm -it ritools-atlas
... use /opt/R-devel/bin/R
